### DISCUSSION TITLE
Please add embeddable gifs

### DISCUSSION THREAD

**Discussion by TodePond** [mozzius/graysky#79] [Feature requests]
Hi it would be great if I could place a URL to a gif in a skeet and then it embeds and autoplays in the client. I think it would be great for graysky users, and also to advance bluesky development on the whole! Also you should come visit the tldraw office some time :)

**Comment by mozzius** [comment]
Totally agree - in retrospect I think my solution is massively over-engineered lol. It'd be great to add more gif providers as well as support gif uploading

re: tldraw office - hell yeah, I'd love to! I hear it's the place to be :)

**Comment by CrosseyeJack** [comment]
Only just noticed gif support was added so I'm late to the party.

I think its a discussion with BlueSky over the "standard" (there is always one more f'ing standard...) would be beneficial. But if we are writing Christmas lists to Santa, I would opt for something along the lines of having some form of header in the server response which then would trigger embedding (and which style to embed), random thing plucked out of no where a x-embed header(s) which can contain the URL to the actual gif/video file and metadata about the file. For example telling the client that this file is muted video which should autoplay (if enabled in the client), looped, with minimal UI on the video (mimicking an embedded gif) or if its a longer form video file which should autoplay (but muted) / shouldn't auto play unmuted, with a video player UI. (But thats getting a bit close to just creating desired settings users pick in their client and i'm rambling...)

Something like this would allow for more media styles to be supported, and to be hosted by any provider using any URL path structure they like, as long as the host serving the file "opts in" by serving content with the correct headers, without the headers the link would just be displayed in the client as if it any other link, Headers could be fetched from the server using a HEAD request as not to pull the whole file (saving bandwidth for both the user and server) and would still allow for the server to serve `content-type text/html` allowing for twitter/og cards to exist (after fetching the page obv) and allow the host to embed the video in a html page for clients that don't support embeds using this method but could still view the gif/video by following the link/card.

As I said at the start, prob a discussion best on the at-proto / bluesky repos but just my 2 pennies on how I would love to see such a system created.
